core/src/network/json_rpc/websocket_session.cpp

```diff
@@ -0,0 +1,119 @@
+// Rewritten and refactored based on code from here:
+// https://github.com/vinniefalco/CppCon2018
+//
+// Copyright (c) 2018 Vinnie Falco (vinnie dot falco at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+
+#include "websocket_session.h"
+
+namespace plc::core::network::json_rpc {
+
+WebSocketCallbacks WebSocketSession::m_callbacks;
+
+WebSocketSession::WebSocketSession(tcp::socket socket, int id)
+    : m_ws(std::move(socket))
+    , m_id(id) {
+    m_log->info("WebSocketSession::WebSocketSession");
+}
+
+WebSocketSession::~WebSocketSession() {
+    m_log->info("WebSocketSession::~WebSocketSession");
+}
+
+void WebSocketSession::fail(error_code ec, char const* what) {
+    m_log->info("WebSocketSession::fail what {}", what);
+    // Don't report these
```

Comment:
The comment says that these (which) errors shouldn't be reported but the code below suggests that espiecially these (still which?) errors like operation aborted or websocket closed will be reported.